src/debugging/netSdk/NetSdkDebugHelper.ts

```diff
@@ -38,14 +53,84 @@ export class NetSdkDebugHelper extends NetCoreDebugHelper {
         return configurations;
     }
 
+    public async afterResolveDebugConfiguration(context: DockerDebugContext, debugConfiguration: DockerDebugConfiguration): Promise<void> {
+        const { task, promise } = netSdkRunTaskProvider.createNetSdkRunTask(
+            {
+                netCore: {
+                    appProject: await this.inferProjPath(context.actionContext, context.folder),
+                }
+            }
+        );
+        await tasks.executeTask(task);
+        await promise;
+    }
+
     /**
-     * Checks if the launch task is using the .NET SDK Container build
-     * @param preLaunchTask
-     * @returns true if the launch task is using the .NET SDK Container build
-     *          false otherwise
+     * @returns the project path stored in the static variable projPath if it exists,
+     *          otherwise prompts the user to select a .csproj file and stores the path
+     *          in the static variable projPath
      */
-    public isDotNetSdkBuild(preLaunchTask: string): boolean {
-        return preLaunchTask === NetSdkTaskFullSymbol;
+    public async inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
```

Comment:
Not strictly necessary, but it can be good to mark the overriding method as `override` to make it more obvious it re-implements a base method.

```suggestion
    public override async inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
```

Reply:
makes sense, i dig it